### PR TITLE
`namespace_linter()` handles backticked operators

### LINKED ISSUE #1752: `namespace_linter()` false positives with re-exported functions
``` r
library(lintr)

dir <- tempfile()
pkg <- usethis::create_package(dir, open = FALSE, rstudio = FALSE)
setwd(pkg)
fs::dir_create("R")
writeLines(
  c("#' @export", "#' @importFrom magrittr \"%>%\"", "magrittr::`%>%`"), 
  "R/script.R"
)
devtools::document()
#> ℹ Updating file6b4e37419a97 documentation
#> ℹ Loading file6b4e37419a97
#> Writing 'NAMESPACE'
#> Writing 'NAMESPACE'
#> Writing 'reexports.Rd'
readLines("NAMESPACE")
#> [1] "# Generated by roxygen2: do not edit by hand"
#> [2] ""                                            
#> [3] "export(\"%>%\")"                             
#> [4] "importFrom(magrittr,\"%>%\")"

devtools::install(quiet = TRUE)
lintr::lint_package(linters = namespace_linter())
#> R/script.R:3:11: warning: [namespace_linter] '`%>%`' is not exported from {magrittr}.
#> magrittr::`%>%`
#>           ^~~~~
```

<sup>Created on 2022-10-23 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Closes #1752

``` r
library(lintr)

lint(
  text = "magrittr::`%>%`",
  linters = namespace_linter()
)

lint(
  text = "magrittr::'%>%'",
  linters = namespace_linter()
)
```

<sup>Created on 2022-10-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: tests/testthat/test-namespace_linter.R [hunk 1]
```diff
@@ -18,6 +18,17 @@ test_that("namespace_linter respects check_exports and check_nonexports argument
...
 })
 
+test_that("namespace_linter respects check_exports and check_nonexports arguments", {
+  skip_if_not_installed("rlang")
+  linter <- namespace_linter()
+
+  expect_lint("rlang::`%||%`", NULL, linter)
+  expect_lint("rlang::`%||%`()", NULL, linter)
+
+  expect_lint("rlang::'%||%'", NULL, linter)
```
Comment (AshesITR) [inline tests/testthat/test-namespace_linter.R:28]: Test `"%>%"` as well just in case?

Comment (IndrajeetPatil) [reply tests/testthat/test-namespace_linter.R:28]: Won't we need to add `{magrittr}` to `Suggests` in that case? To make that namespace available?

Comment (AshesITR) [reply tests/testthat/test-namespace_linter.R:28]: Why is that? We can use `rlang` for that as well. It's just a different way of quoting `%>%`.

Comment (IndrajeetPatil) [reply tests/testthat/test-namespace_linter.R:28]: I am not sure what you mean: `%>%` is not exported from `{rlang}`.

Comment (IndrajeetPatil) [reply tests/testthat/test-namespace_linter.R:28]: @AshesITR I've added a test, well sort of. See if it works for you.

Comment (AshesITR) [reply tests/testthat/test-namespace_linter.R:28]: Now I'm completely confused 😅
Shouldn't the linter fire for non-exported objects, i.e. the tests fail?
I'll look at it in more detail later.

Comment (IndrajeetPatil) [reply tests/testthat/test-namespace_linter.R:28]: This is the existing behaviour, and is unrelated to the current PR. Linter doesn't fail for non-exported objects, just produces a lint.

https://github.com/r-lib/lintr/blob/1ac48e94ef550945d94c0e47b43a9eb8041b5f00/tests/testthat/test-namespace_linter.R#L30-L34

Comment (AshesITR) [reply tests/testthat/test-namespace_linter.R:28]: Ah, my bad. I just noticed `%||%` is tested, not `%>%`.
My original remark was supposed to be test `rlang::"%||%"` in addition to the other quoting options.
Sorry for the confusion!

Comment (IndrajeetPatil) [reply tests/testthat/test-namespace_linter.R:28]: I see. So, in addition to `rlang::'%||%'`, which is already covered, you want me to check `rlang::"%||%"`.

Can do.

**Review by AshesITR — APPROVED** [review]
Thanks!

**Pull request merged**